luigi/cmdline_parser.py

```diff
@@ -130,6 +132,6 @@ def _possibly_exit_with_help(parser, known_args):
         """
         Check if the user passed --help, if so, print a message and exit.
         """
-        if known_args.core_help:
+        if known_args.core_help or known_args.core_help_all:
             parser.print_help()
             raise SystemExit('Exiting due to --help was passed')
```

Comment:
This message is incorrect now ("--help-all" != "--help") and I think no message should be printed at all. Spammy unlike any other program I've seen. I find it hardly likely for a person to need an explanation as to why the program printed something that looked like help and exited.